app/scripts/controllers/reports/aboRhGroupsReport.js

```diff
@@ -23,6 +23,16 @@ angular.module('bsis')
       $scope.submitted = false;
     };
 
+    function createEmptyRow() {
+
+      var emptyRow = {};
+      emptyRow.cohorts = [{}, {option:''}, {option:''}];
+      emptyRow.venue = {};
+      emptyRow.venue.name = '';
+      return emptyRow;
+
+    }
+
```

Comment:
If possible, we should avoid adding a blank row. What I'd suggest is extracting the logic from the forEach loop into a function and calling that function an extra time after the loop finishes. Or using a different loop construct (e.g. do while) where the code can be executed an extra time - after all the rows have been processed.